### ISSUE TITLE
variables data repeat rendering 

### ISSUE THREAD

**Issue by Gc-Peanut** [maizzle/framework#1112]
- Maizzle Version: 4.6.1
- Node.js Version: 20.10.0
![image](https://github.com/maizzle/framework/assets/147602061/f1da8476-c244-4fca-ad7b-aefe76678a9f)

```
const financeColumns = [
    {
        title: "币种",
        key: "coin",
        width: "56px",
    },
    {
        title: "网络",
        key: "network",
        width: "80px",
    },
    {
        title: "总金额",
        key: "amount",
    },
    {
        title: "钱包总余额",
        key: "total",
    },
]
```

```
module.exports = {
    build: {
        templates: {
            source: 'src/templates',
            destination: {
                path: 'build_local',
            },
            assets: {
                source: 'src/images',
                destination: 'images',
            },
        },
    },
    locals: {
        finance,
    }
}
```

template

```
<x-table columns="{{finance.columns}}" />
```

**Comment by Gc-Peanut** [comment]
<script props>
    const TD_CLASS_NAME = "px-4 py-3 leading-4.5 border-solid border-b border-l-0 border-r-0 border-t-0 border-gary-200";
    module.exports = {
        columns: props.columns || [
  { title: 'Name', key: 'name', width: "56px" },
  { title: 'Age', key: 'age' },
  // Add more columns as needed
],
        data: props.data || [
  { name: 'John Doe', age: 30 },
  { name: 'Jane Doe', age: "w-full box-border text-left text-xs rounded border-collapse border-spacing-0 border border-solid border-gary-200" },
  // Add more data rows as needed
],
        showHeader: props.showHeader || true,
        TD_CLASS_NAME,

    }
  </script>
 
<table class="w-full box-border text-black-400 text-left text-xs rounded border-spacing-0 border border-solid border-gary-200 overflow-hidden">
    <if condition="showHeader">
        <thead class="bg-slate-50">
            {{columns}}
            <tr>
                <for loop="col, colIndex in columns">
                    <if condition="col.width">
                        <th class="{{TD_CLASS_NAME}} text-black-300 font-normal" style="width: {{col.width}};">{{col.title}}</th>
                    </if>
                    <else>
                        <th class="{{TD_CLASS_NAME}} text-black-300 font-normal" >{{col.title}}</th>
                    </else>
                </for>
            </tr>
        </thead>
    </if>
    <tbody>
        <for loop="row, rowIndex in data">
            <tr>
                <for loop="col, colIndex in columns">
                    
                    <if condition="rowIndex === (data.length - 1)">
                        <if condition="col.width">
                            <td class="{{TD_CLASS_NAME}} border-b-0" style="width: {{col.width}};">{{row[col.key]}}</td>
                        </if>
                        <else>
                            <td class="{{TD_CLASS_NAME}} border-b-0">{{row[col.key]}}</td>
                        </else>
                    </if>
                    <else>
                        <if condition="col.width">
                            <td class="{{TD_CLASS_NAME}}" style="width: {{col.width}};">{{row[col.key]}}</td>
                        </if>
                        <else>
                            <td class="{{TD_CLASS_NAME}}">{{row[col.key]}}</td>
                        </else>
                    </else>
                </for>
            </tr>
        </for>
    </tbody>
</table>

**Comment by Gc-Peanut** [comment]
compoents
[table.txt](https://github.com/maizzle/framework/files/13699230/table.txt)
后缀改文 html 使用

**Comment by cossssmin** [comment]
You're defining `financeColumns` and never using it, instead you're using `finance.columns` which in your example would be undefined.

Btw you can use `financeColumns` directly in the `<x-table>` component if you do this:

```js
module.exports = {
  // ...
  locals: {
    financeColumns
  }
}
```

**Comment by Gc-Peanut** [comment]
![image](https://github.com/maizzle/framework/assets/147602061/552557f3-2c00-483d-ab74-10f611d320cd)
![image](https://github.com/maizzle/framework/assets/147602061/0b6911cf-be59-4976-9c04-8dc030a135f7)
![image](https://github.com/maizzle/framework/assets/147602061/b52273c9-7464-4b2c-a949-73a79ada3e33)

**Comment by cossssmin** [comment]
Note that there's currently a [bug with object merging](https://github.com/maizzle/framework/issues/1057#issuecomment-1872076756) that duplicates data, will have it fixed shortly.

**Comment by cossssmin** [comment]
Should be fixed in `v4.6.3` but let me know if you're still having issues with this.

https://github.com/maizzle/framework/releases/tag/v4.6.3
